core/batcher.go

```diff
@@ -0,0 +1,348 @@
+package core
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/anoideaopen/foundation/core/cachestub"
+	"github.com/anoideaopen/foundation/core/telemetry"
+	"github.com/anoideaopen/foundation/core/types"
+	"github.com/anoideaopen/foundation/hlfcreator"
+	"github.com/anoideaopen/foundation/internal/config"
+	"github.com/anoideaopen/foundation/proto"
+	pb "github.com/golang/protobuf/proto" //nolint:staticcheck
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	BatcherBatchExecuteEvent                    = "batcherBatchExecute"
+	TxBatcherRequestType     BatcherRequestType = "tx"
+)
+
+type (
+	BatcherBatchExecuteRequestDTO struct {
+		Requests []BatcherRequest `json:"requests"`
+	}
+
+	BatcherRequestType string
+
+	BatcherRequest struct {
+		BatcherRequestID   string             `json:"batcher_request_id"`   // BatcherRequestID batcher request id
+		Channel            string             `json:"channel"`              // Channel on which the chaincode will be invoked
+		Chaincode          string             `json:"chaincode"`            // Chaincode of the chaincode to invoke
+		Method             string             `json:"function"`             // Method of the chaincode function to invoke
+		Args               []string           `json:"args"`                 // Args to pass to the chaincode function
+		BatcherRequestType BatcherRequestType `json:"batcher_request_type"` // BatcherRequestType for choose method to handling request. for example tx, swaps, swaps_keys, multi_swaps, multi_swaps_keys
+	}
+
+	Batcher struct {
+		batchCacheStub *cachestub.BatchCacheStub
+		cc             *ChainCode
+		cfgBytes       []byte
+		ski            string
+		tracingHandler *telemetry.TracingHandler
+	}
```

Comment:
All fields in the Batcher struct are private, and it’s unclear where the methods are because the code is in a large type block. We need to declare this either along with the methods or make the fields public.

https://google.github.io/styleguide/go/best-practices

Reply:
@muzykantov I did it, thank you